evalml/utils/woodwork_utils.py

```diff
@@ -98,3 +100,23 @@ def _retain_custom_types_and_initalize_woodwork(old_datatable, new_dataframe, lt
             except (ValueError, TypeError):
                 pass
     return ww.DataTable(new_dataframe, logical_types=retained_logical_types)
+
+
+def _convert_numeric_dataset_for_data_sampler(X, y, to_pandas=True):
```

Comment:
@bchen1116 could you please split this into two methods instead of using this `to_pandas` flag? Its a creative solution, but I think the code will be easier to understand and debug if our methods have one possible input type and one possible output type.